Final chapters of Fundamentals of Image and Imaging (#162)
johko/computer-vision-course: chapters/en/Unit 1 - Fundamentals/image_and_imaging/examples-preprocess.mdx

# Pre-processing for Computer Vision Tasks

Now that we have seen what are images, how they are acquired, and their impact, it is time to understand what operations we can perform and how they are used during the model-building process. 

## Operations in Digital Image Processing
In digital image processing, operations on images are diverse and can be categorized into:

- Logical
- Statistical
- Geometrical
- Mathematical
- Transform operations.
 

> Each category encompasses different techniques, such as morphological operations under logical operations and fourier transforms and principal component analysis (PCA) under transforms. In this context, we refer to morphology as the group of operations tha use structuring elements to generate images of the same size by looking into the values of the pixel neighborhood. Understanding the distinction between element-wise and matrix operations is important in image manipulation. Element-wise operations, such as raising an image to a power or dividing it by another image, involve processing each pixel individually. This pixel-based approach contrasts with matrix operations, which utilize matrix theory for image manipulation. Having said that, you can do whatever you want with images, as they are matrices containing numbers!

Reviewer: getting a bit confused by the many "and"s in the first sentence, maybe it can be replaced as in my suggestion
```suggestion
Each category encompasses different techniques, such as morphological operations under logical operations or fourier transforms and principal component analysis (PCA) under transforms. In this context, we refer to morphology as the group of operations tha use structuring elements to generate images of the same size by looking into the values of the pixel neighborhood. Understanding the distinction between element-wise and matrix operations is important in image manipulation. Element-wise operations, such as raising an image to a power or dividing it by another image, involve processing each pixel individually. This pixel-based approach contrasts with matrix operations, which utilize matrix theory for image manipulation. Having said that, you can do whatever you want with images, as they are matrices containing numbers!
```

> # Examples of pre-processing for Computer Vision

Reviewer 2: ```suggestion
# Pre-processing for Computer Vision Tasks
```

> Now that we have seen what are images, how they are acquired and their impact, it is time to understand what are the operations we can perform them and how they are used during the model building process. 

Reviewer 2: ```suggestion
Now that we have seen what are images, how they are acquired, and their impact, it is time to understand what operations we can perform and how they are used during the model-building process. 
```

## Operations in Digital Image Processing
In digital image processing, operations on images are diverse and can be categorized into:

- Logical
- Statistical
- Geometrical
- Mathematical
- Transform operations.
 

> Each category encompasses specific techniques, such as Morphology under Logical Operations and Fourier and Principal Component Analysis (PCA) under Transforms. Understanding the distinction between elementwise and matrix operations is important in image manipulation. Element wise operations, such as raising an image to a power or dividing it by another image, involve processing each pixel individually. This pixel-based approach contrasts with matrix operations, which utilize matrix theory for image manipulation. Having said that, you can do whatever you want with images, as they are matrices containing numbers!

Reviewer 2: ```suggestion
Each category encompasses different techniques, such as morphological operations under logical operations and fourier transforms and principal component analysis (PCA) under transforms. Understanding the distinction between element-wise and matrix operations is important in image manipulation. Element-wise operations, such as raising an image to a power or dividing it by another image, involve processing each pixel individually. This pixel-based approach contrasts with matrix operations, which utilize matrix theory for image manipulation. Having said that, you can do whatever you want with images, as they are matrices containing numbers!
```
Is morphology a technique though, isn't it more like a layer?

Author: Hm, not in this context. Classically speaking, morphological operations are the group of techniques where you pass a structuring element in an image and normally the image stays about the same size. I think the classical example is erosion and dilation. In my experience, they are really common when you have the end users til interacting with the image or some human-in-the-loop. Sometimes the users are not happy about the segmentation mask (either because they are disconnected, or too small, or too big) so let them correct it using these morphological operations instead of trying to re-tune or re-run the model.

Author: @krmdel @merveenoyan  I've added this clarification on text.

## Mathematical Tools in Image Processing

> 

Reviewer 2: ```suggestion
```

> Mathematical tools are indispensable in digital image processing. Set theory, for instance, is crucial for understanding and performing operations on images, particularly binary images. In these images, pixels are typically categorized as either foreground (1) or background (0). Operations such as union and intersection in set theory are used to determine relationships between features represented by pixel coordinates. Intensity transformations and spatial filtering are other mathematical tools. They focus on manipulating pixel values within an image, where operators are applied to single images or a set of images for various purposes, like noise reduction.

Reviewer 2: ```suggestion
Mathematical tools are indispensable in digital image processing. Set theory, for instance, is crucial for understanding and performing operations on images, particularly binary images. In these images, pixels are typically categorized as either foreground (1) or background (0). In set theory, operations such as union and intersection determine relationships between features represented by pixel coordinates. Intensity transformations and spatial filtering are other mathematical tools. They focus on manipulating pixel values within an image, where operators are applied to single images or a set of images for various purposes, like noise reduction.
```

> 

Reviewer 2: Maybe you could also mention the most used transformations like contrast stretching or histogram matching

Author: @krmdel this is an example of one where you can edit in the doc first and then change it on the .mdx file latter :)

## Spatial Filtering Techniques and Image Enhancement

> Spatial filtering encompasses a broad range of applications in image processing, primarily focused on modifying images by altering each pixel's value based on its neighbouring pixels' values. Techniques include linear spatial filters, which can either blur (lowpass filters) or sharpen (high pass filters) an image. The properties and applications of different filter kernels, such as the Gaussian filter and box filters, are contrasted. Sharpening filters, which emphasize transitions in intensity, are often implemented through digital differentiation techniques like the Laplacian, highlighting edges and discontinuities in an image.

Reviewer 2: ```suggestion
Spatial filtering encompasses a broad range of applications in image processing, primarily modifying images by altering each pixel's value based on its neighboring pixels' values. Techniques include linear spatial filters, which can blur (low pass filters) or sharpen (high pass filters) an image. The properties and applications of different filter kernels, such as the Gaussian and box filters, are contrasted. Sharpening filters emphasize transitions in intensity and are often implemented through digital differentiation techniques like the Laplacian, highlighting edges and discontinuities in an image.
```
